vertical-pod-autoscaler/pkg/utils/vpa/capping.go

```diff
@@ -62,7 +65,11 @@ func (c *cappingRecommendationProcessor) Apply(
 	}
 	updatedRecommendations := []vpa_types.RecommendedContainerResources{}
 	containerToAnnotationsMap := ContainerToAnnotationsMap{}
-	for _, containerRecommendation := range podRecommendation.ContainerRecommendations {
+	limitAdjustedRecommendation, err := c.capProportionallyToMaxLimit(podRecommendation, pod, containerToAnnotationsMap)
```

Comment:
This doesn't work. We need to apply capping also to upper and lower bound. What we need is to integrate capping to max/min limitrange into getCappedRecommendationForContainer function. 
Just as we have ApplyVpaPolicy, we need to have ApplyMaxRange here:
https://github.com/kubernetes/autoscaler/blob/5094dc06c0bdfc50a543e4ae719c4427dcf029d3/vertical-pod-autoscaler/pkg/utils/vpa/capping.go#L146

Apologies for not spotting it earlier.

Reply:
Changed to adjust not only target but also bounds. I think this will work correctly (if limit range, limit/request ratio and policy allow assigning a recommendation).